### ISSUE TITLE
GPU drivers aren't open.

### ISSUE THREAD

**Issue by jackffmm** [starfive-tech/soft_3rdpart#4]
Why you dont release the source code of the gpu drivers binary ? You are slowing the sales of your product.
First because a lot of people dont trust the blob. Second because the community can help and improve the drivers.

Thanks.

**Comment by coderamen666** [comment]
The GPU and the drivers are licensed from Imagination Technologies. Imagination is trying to write an open mesa driver that supports the VF2. This is something temporary that StarFive doesn't have the power to do. This is not an ideal situation, agreed, but you just have to wait. It is exactly like the Raspberry Pi's situation a few years ago with their proprietary stack from Broadcom. It will take time, but this is a known issue.

**Comment by jackffmm** [comment]
why the open drivers should be different from the licensed drivers ?

**Comment by coderamen666** [comment]
I get how you feel. I wish that NDAs and corporate IP secrecy was less prevalent as well. It's just that corporations and their fondness of intellectual property suck. That's your answer.

**Comment by jackffmm** [comment]
but the owner of the IP can release the code

**Comment by coderamen666** [comment]
Yeah the owner of the IP should do it but they won't unless there is some kind of pressure.

**Comment by jackffmm** [comment]
theyr loss

**Comment by coderamen666** [comment]
I agree with you it's just corporations are shitty.

**Comment by coderamen666** [comment]
Ok bud I got my board now and I heard they are working on a Vulkan driver that will be useful in this effort.

**Comment by intx82** [comment]
Hi guys, does anyone tried to use PVR kernel-module from MediaTek? Seems they should be compatible with our closed-source firmware (1.15) and works on VF2

[https://github.com/PQEnablers-Devices/android_kernel_xiaomi_mt6877/tree/719e9bfb4d55716158a51911f6e370a08a97a532/drivers/gpu/mediatek/gpu_rgx/m1.15ED5904022](https://github.com/PQEnablers-Devices/android_kernel_xiaomi_mt6877/tree/719e9bfb4d55716158a51911f6e370a08a97a532/drivers/gpu/mediatek/gpu_rgx/m1.15ED5904022)

**Comment by coderamen666** [comment]
I thought that the out of tree kernel build used by SiFive already had an open kernel module. This thread is mainly concerned with userspace drivers.

**Comment by jackffmm** [comment]
I knew that the open drivers are 2d only

**Comment by anoduck** [comment]
This is not what I invested in when I purchased my device.
